### ISSUE TITLE
Make CsvRelation to extend HadoopFsRelation

### ISSUE THREAD

**Issue by zjffdu** [databricks/spark-csv#197]
CsvRelation could extend HadoopFsRelation to leverage the features in HadoopFsRelation. I am working on SPARK-11691 to allow HadoopFsRelation to support compression which CsvRelation requires. After SPARK-11691 is done, I will do this ticket.

**Comment by HyukjinKwon** [comment]
Is this a duplicate (or subset) of #123 ?

**Comment by zjffdu** [comment]
Yes, after CsvRelatioin extends HadoopFsRelation, partition will be supported

**Comment by HyukjinKwon** [comment]
I am leaving this link for a reminder that `HadoopFSRelation` is changed.
https://issues.apache.org/jira/browse/SPARK-13664
